flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/TableWithSQLITCase.scala

```diff
@@ -113,4 +114,26 @@ class TableWithSQLITCase(
     val results = result.toDataSet[Row].collect()
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
+
+  @Test
+  def testUDTFWithPojoType(): Unit = {
```

Comment:
We should keep ITCases to a very minimum, because they are very expensive. Espacially, we should not add 4 ITCases for a one line change. I think it is sufficient if we just extend an existing test in `DataSetUserDefinedFunctionITCase` and `DataStreamUserDefinedFunctionITCase`. You could e.g. rename `testLongAndTemporalTypes` to `testDifferentTypes` and add a POJO there.